packages/eslint-plugin/tests/rules/prefer-find.test.ts

```diff
@@ -0,0 +1,532 @@
+import { noFormat, RuleTester } from '@typescript-eslint/rule-tester';
+
+import rule from '../../src/rules/prefer-find';
+import { getFixturesRootDir } from '../RuleTester';
+
+const rootDir = getFixturesRootDir();
+
+const ruleTester = new RuleTester({
+  parser: '@typescript-eslint/parser',
+  parserOptions: {
+    tsconfigRootDir: rootDir,
+    project: './tsconfig.json',
+  },
+});
+
+ruleTester.run('prefer-find', rule, {
+  valid: [
+    `
+      interface JerkCode<T> {
+        filter(predicate: (item: T) => boolean): JerkCode<T>;
+      }
+
+      declare const jerkCode: JerkCode<string>;
+
+      jerkCode.filter(item => item === 'aha');
```

Comment:
[Testing] This should have a `[0]` at the end, as the current code wouldn't violate the rule.

But otherwise +1 😄 love the copying of the code example from before!

Reply:
good catch. changed!